### DISCUSSION TITLE
Artillery-Playwright (with POM and Fixture)

### DISCUSSION THREAD

**Discussion by ldelreguero** [artilleryio/artillery#3327] [Artillery]
Hello Community!

I implemented (I think) Playwright with POM and Fixture.

Now, I want to try Artillery.

I have the following structure:

logis.js

![image](https://github.com/user-attachments/assets/06c0e25d-0309-4654-9aa9-7c9b239b0896)

fixture.js

![image](https://github.com/user-attachments/assets/d8b57169-39fd-461b-8b83-bd511ac48c2f)

page1,js

![image](https://github.com/user-attachments/assets/847e9a12-d7f5-4bb4-9a6f-5d4dd0ae8c87)

Test in Playwright:

![image](https://github.com/user-attachments/assets/3fc42af5-a7b9-4a00-bdfa-ef7e11dc1904)

Test in Artillery:

![image](https://github.com/user-attachments/assets/0eb16b22-3ae8-4c2c-9e32-61330ad592d0)

yml:

![image](https://github.com/user-attachments/assets/41d9b475-52cc-4cbe-921e-7346550dffec)

Ejecution:

![image](https://github.com/user-attachments/assets/c7c84d00-2562-427f-9445-dc4425d7e04d)

Does Artillery support Playwright's Fixture concept?

What could I be doing wrong in the Artillery Test? Playwright's test is correct but Artillery's is not

Is it a configuration issue? (yml)

I look forward to any comments. Everything will be very useful for me.

**Comment by hassy** [comment]
hi @ldelreguero 👋 Artillery's `testFunction` functions take `page` as the first argument. In your code you're destructuring that argument into `webApp` and `page` variables, so they end up being `undefined`.

Change your code like below and it should work:

```diff
- async function helloFlow({ webApp, page }) {
+ async function helloFlow(page) {
```

**Reply by ldelreguero** [reply]
You are a genius!
Thank you very much!

I confirm that, at least for now, to test artillery:
I am not using "fixture" from Playwright and I have tested this function.
It is a small modification, but very useful!

Thank you very much again

**Reply by dmelo7** [reply]
@ldelreguero did you manage to use fixtures on artillery? is that possible? thanks!
